### PR TITLE
feat: Set up kustomization

### LINKED ISSUE #448: Add K8s Kustomization
### Description

To replace the existing Helm chart

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by microamp**
Fixes #448 

Deleted the existing `k8s` directory before adding kustomization. 

The deployment file is based on the existing Helm chart. e.g.

```sh
helm install bib helm --dry-run > helm-install-dry-run-output.yaml
```

Further checks and cleanups may be desired.

### CODE REVIEW

**Comment by microamp** [comment]
@kesara @NGPixel Switched to Review to review to get some feedback.

File: k8s/django-config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  PORT: "8000"
+  PRIMARY_HOSTNAME: "bib.ietf.org"
+  PYTHONUNBUFFERED: "1"
+  REDIS_HOST: "localhost"
+  REDIS_PORT: "6379"
+  SENTRY_DSN: ""
+  SERVER_EMAIL: "support@ietf.org"
+  SERVICE_NAME: "IETF BibXML Service"
+  SOURCE_REPO_URL: "https://github.com/ietf-tools/bibxml-service"
+  SNAPSHOT: "1.3.0"
```
Comment (kesara) [inline k8s/django-config.yaml:26]: Can we use `APP_IMAGE_TAG` for this?

File: k8s/bib.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            capabilities:
+              drop:
+              - ALL
+            readOnlyRootFilesystem: true
+            runAsUser: 1000
+            runAsGroup: 1000
+          command:
+            - /bin/sh
+            - -c
+            - |
```
Comment (NGPixel) [inline k8s/bib.yaml:71]: Omit shell

**Review by kesara — APPROVED** [review]

**Comment by microamp** [comment]
Update: Rebased to apply https://github.com/ietf-tools/bibxml-service/pull/451 to this PR.

File: k8s/django-config.yaml (hunk 1 above)
Comment (microamp) [reply k8s/django-config.yaml]: Config map has been deprecated in favour of the external secret, hence resolved.

File: k8s/bib.yaml [hunk 3]
```diff
@@ -91,6 +93,9 @@ spec:
         - name: datasets
           emptyDir:
             sizeLimit: 5Gi
+        - name: tmp
+          emptyDir:
+            sizeLimit: 1Gi
```
Comment (microamp) [inline k8s/bib.yaml:98]: @kesara This was to fix 
```
[2024-08-07 21:48:12,394: CRITICAL/MainProcess] Unrecoverable error: FileNotFoundError(2, "No usable temporary directory found in ['/tmp', '/var/tmp', '/usr/tmp', '/code']")
```

Let me know if `1Gi` is overkill.

**Review by NGPixel — APPROVED** [review]

**Pull request merged**